### DISCUSSION TITLE
Query on behaviour when --branch switch used

### DISCUSSION THREAD

**Discussion by gb250155** [zricethezav/gitleaks#593] [Q&A]
Hi, I'm writing a Powershell script to make use of Gitleaks in one of our Jenkins build scan jobs.

I've been using both the --branch and --commit-since switches on one of our test repositories to test the functionality and ensure our setup is OK, and in doing so I've noticed some odd behaviour.

The background to this is that the repo in question had a hard-coded secret committed back in April 2021, so I would expect that to show up in a scan of the entire repository. I'm operating on the "main" branch which is our default branch.

Using those two switches in the command, the following results occur:

* --branch switch used for "main" branch: commit with hard-coded secret found, this is as expected
* --commit-since used to cover commits from past 7 days: commit with hard-coded secret not found, this is as expected
* --branch and --commit-since switches used together with values as above: commit with hard-coded secret found, NOT as expected

Are these two switches supposed to be mutually exclusive? Or does the --branch switch scan the entire branch by default and ignores the --commit-since switch?

Many thanks,
Geoff

**Comment by w0rmr1d3r** [comment]
Hello @gb250155 !
I don't know if they are mutually exclusive, but it seems so.

Shall we get the documentation for those options updated?

Cheers!
